context.go

```diff
@@ -94,13 +94,15 @@ func (p *parseContext) Stop(err error, branch *parseContext) bool {
 		p.deepestError = err
 		p.deepestErrorDepth = maxInt(branch.PeekingLexer.Cursor(), branch.deepestErrorDepth)
 	}
-	if branch.PeekingLexer.Cursor() > p.PeekingLexer.Cursor()+p.lookahead {
+	if !p.hasInfiniteLookahead() && branch.PeekingLexer.Cursor() > p.PeekingLexer.Cursor()+p.lookahead {
```

Comment:
The impetus here is that sometimes, a grammar needs infinite lookahead. This becomes a bit more clear with the introduction of unions. See comments in `parser_advanced_test.go` for a more specific discussion of this